runner/runners/invoke.go

```diff
@@ -47,58 +47,60 @@ func (inv *Invoker) Run(cmd *runner.Command, id runner.RunID) (abortCh chan<- st
 	return abortChFull, updateChFull
 }
 
-type checkoutAndError struct {
-	checkout snapshot.Checkout
-	err      error
-}
-
 // Run runs cmd as run id returning the final ProcessStatus
 // Run will send updates the process is running to updateCh.
 // Run will enforce cmd's Timeout, and will abort cmd if abortCh is signaled.
 // Run will not return until the process is not running.
+// NOTE: kind of gnarly since our filer implementation (gitdb) currently mutates the same worktree on every op.
 func (inv *Invoker) run(cmd *runner.Command, id runner.RunID, abortCh chan struct{}, updateCh chan runner.RunStatus) (r runner.RunStatus) {
 	log.Printf("run. id: %v, cmd: %+v", id, cmd)
 	defer func() {
 		updateCh <- r
 		close(updateCh)
 	}()
 
-	checkoutCh := make(chan checkoutAndError)
-	var checkout snapshot.Checkout
-	var err error
+	var co snapshot.Checkout
+	checkoutCh := make(chan error)
 	go func() {
 		if cmd.SnapshotID == "" {
-			//TODO: we don't want this logic to live here, these decisisions should be made at a higher level.
+			//TODO: we don't want this logic to live here, these decisions should be made at a higher level.
```

Comment:
I agree we should require snapshotId.   The execers.UseSimExecerArg test brings special test behavior into the code - This is an anti-pattern. We should update the tests to have a noop checkouter when SimExecer is used and provide a dummy snapshotId.  (if this can't be done now, we should create a techdebt jira issue and put the issue number in the comment).

Also will we have a task with no cmd.Argv?